apps/C/breakout.c

```diff
@@ -0,0 +1,222 @@
+            raw = raw + 2 ;
+            if(raw == (RAWS * 2)) /* End of screen */
+            {
+                raw = 0;
+            }
+            i++;
+            continue;
+        }
+
+        draw_char(c[i], raw, col);
+        col++;
+        if(col == COLUMN) /* End of line */
+        {
+            col = 0;
+            raw = raw + 2 ;
+            if(raw == (RAWS * 2)) /* End of screen */
+            {
+                raw = 0;
+            }
+        }
+        i++;
+    }
+    
+    /* Update CR_CURSOR */
+    WRITE_REG(CR_CURSOR_H, col);
+    WRITE_REG(CR_CURSOR_V, raw);
+}
+
+/* This function print a symbol from anime table on the screen in (raw,col) position */
+void draw_symbol(int symbol, int raw, int col)
+{
+    unsigned int vertical   = raw * (2*LINE);
+    unsigned int horizontal = col * BYTES;
+    volatile int *ptr_top;
+    volatile int *ptr_bottom;
+
+    VGA_PTR(ptr_top    , horizontal + vertical);
+    VGA_PTR(ptr_bottom , horizontal + vertical + LINE);
+
+    WRITE_REG(ptr_top    , ANIME_TOP[symbol]);
+    WRITE_REG(ptr_bottom , ANIME_BOTTOM[symbol]);
+}
+
+/* This function clear the screen */
+void clear_screen() 
+{
+    int i = 0;
+    volatile int *ptr;
+    VGA_PTR(ptr , 0);
+    for(i = 0 ; i < VGA_MEM_SIZE_WORDS ; i++)
+    {
+        ptr[i] = 0;
+    }
+}
+
+void set_cursor(int raw, int col)
+{
+    WRITE_REG(CR_CURSOR_H, col);
+    WRITE_REG(CR_CURSOR_V, raw);
+}
+
+void draw_horizontal_line(int start, int end, int raw)
+{
+    for(int i = start ; i < (end + 1) ; i++)
+    {
+        draw_symbol(5, raw, i);
+    }
+}
+
+void clear_horizontal_line(int start, int end, int raw)
+{
+    for(int i = start ; i < (end + 1) ; i++)
+    {
+        draw_symbol(6, raw, i);
+    }
+}
+
+void draw_vertical_line(int start, int end, int col)
+{
+    for(int i = start ; i < (end + 1) ; i++)
+    {
+        draw_symbol(5, i, col);
+    }
+}
+
+struct point {
+    int pos_x;
+    int pos_y;
+};
+
+struct game_tool {
+    struct point center;
+    int start;
+    int end;
+};
+
+void set_gt(struct game_tool *gt, int pos_x, int pos_y) 
+{
+    gt->center.pos_x = pos_x;
+    gt->center.pos_y = pos_y;
+    gt->start = pos_x - 5;
+    gt->end   = pos_x + 5;
+}
+
+void display(struct game_tool gt) 
+{
+    draw_horizontal_line(gt.start,gt.end,gt.center.pos_y);
+}
+
+void clear(struct game_tool gt) 
+{
+    clear_horizontal_line(gt.start,gt.end,gt.center.pos_y);
+}
+
+void move_left(struct game_tool *gt)
+{
+    if((gt->start - 1) == 0){
+        return;
+    }
+    else{
+        set_gt(gt, gt->center.pos_x - 1, gt->center.pos_y);
+    }
+}
+
+void move_right(struct game_tool *gt)
+{
+    if((gt->end + 1) == 60){
+        return;
+    }
+    else{
+        set_gt(gt, gt->center.pos_x + 1, gt->center.pos_y);
+    }
+}
+
+void delay(int num) {
+    int counter = 0;
+    int i = 0;
+    for (int i = 0; i < num; i++){counter++;}
+    return;
+}
+
+int main() {
+
+    clear_screen();
+
+    /* Create the board game */
```

Comment:
I'm not an SW expert - 
But should we have a function called "draw_board_game" and have all this there?